### PR TITLE
Last part of improvements related to new click method and our general style

### LINKED ISSUE #2562: Refactor remaining UI modules to use click method + Get rid of *_json() methods + Improve overall style
#2556 introduces the click method to reduce the amount of code needed for clicking on UI elements. In that PR only the ```ui.activationkey``` module is being refactored just as a proof of concept, the remaining modules should be refactored when that PR is merged.
Also, we have newer style to work with API entities from Nailgun, so we should change our code correspondingly. (e.g. change 'create_json()' on 'create()')
Also, we need to improve our code style (working with ddt, method names, formatting and etc)

Full list of modules:
* [ ] `ui.activationkey`
* [x] `ui.architecture`
* [x] `ui.computeresource`
* [x] `ui.configgroups`
* [x] `ui.contentenv`
* [x] `ui.contentsearch`
* [x] `ui.contentviews`
* [ ] `ui.discoveryrules`
* [x] `ui.domain`
* [x] `ui.environment`
* [x] `ui.gpgkey`
* [x] `ui.hardwaremodel`
* [x] `ui.hostgroup`
* [x] `ui.hosts`
* [ ] `ui.ldapauthsource`
* [x] `ui.location`
* [x] `ui.login`
* [x] `ui.medium`
* [x] `ui.navigator`
* [x] `ui.operatingsys`
* [x] `ui.org`
* [x] `ui.partitiontable`
* [x] `ui.products`
* [x] `ui.puppetclasses`
* [x] `ui.repository`
* [x] `ui.role`
* [x] `ui.settings`
* [x] `ui.subnet`
* [x] `ui.subscription`
* [x] `ui.sync`
* [x] `ui.syncplan`
* [x] `ui.template`
* [x] `ui.trend`
* [x] `ui.user`
* [x] `ui.usergroup`

### PR DESCRIPTION
**Pull request by oshtaier**
That is last piece of code related to and Closed #2562.
Also, removing completely redundant tests and lowering coverage a bit (which is redundant from my point of view as well). Removing useless text in test descriptions
```
nosetests tests/foreman/ui/test_ldapauthsource.py 
..............
----------------------------------------------------------------------
Ran 14 tests in 262.340s

OK
```
```
nosetests tests/foreman/ui/test_discoveryrule.py
..............................
----------------------------------------------------------------------
Ran 30 tests in 679.930s

OK
```
```
nosetests tests/foreman/ui/test_activationkey.py
E....SE.S...S.........S...E.......E
----------------------------------------------------------------------
Ran 35 tests in 3575.817s

FAILED (SKIP=4, errors=4)
```
Few failures related to tests which are working with VM (which is not configured on my env), but it doesn't seem that I introduced something harmful here

### CODE REVIEW

File: tests/foreman/ui/test_activationkey.py [hunk 1]
```diff
@@ -211,378 +158,236 @@ def test_positive_create_activation_key_3(self, env_name):
...
-                tab_locators['ak.host_collections.list']
-            ).click()
+            self.activationkey.click(tab_locators['ak.host_collections.list'])
             strategy, value = tab_locators['ak.host_collections.add.select']
             host_collection = self.activationkey.wait_until_element(
-                (strategy, value % host_col['name']))
+                (strategy, value % host_col.name))
             self.assertIsNotNone(host_collection)
 
-    def test_positive_create_activation_key_6(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_activationkey.py]: That is a default scenario which is definitely covered in scope of basic scenario `test_positive_create_activation_key_with_different_names`

File: tests/foreman/ui/test_activationkey.py [hunk 2]
```diff
@@ -211,378 +158,236 @@ def test_positive_create_activation_key_3(self, env_name):
...
-                org=self.org_name,
+                org=self.organization.name,
                 name=name,
                 env=ENVIRONMENT,
-                limit=limit,
-                description=description
-            )
-            self.assertIsNotNone(self.activationkey.search_key(name))
-
-    def test_positive_create_activation_key_8(self):
```
Comment (oshtaier) [inline tests/foreman/ui/test_activationkey.py]: That is a default scenario which is definitely covered in scope of basic scenario `test_positive_create_activation_key_with_different_names`

File: tests/foreman/ui/test_activationkey.py [hunk 3]
```diff
@@ -211,378 +158,236 @@ def test_positive_create_activation_key_3(self, env_name):
...
                 env=ENVIRONMENT,
-                description=gen_string('alpha', 16)
+                description=gen_string('utf8'),
             )
             self.assertIsNotNone(self.activationkey.search_key(name))
-            self.activationkey.delete(name, True)
-            self.assertIsNone(self.activationkey.search_key(name))
-
-    @data(*valid_data_list())
-    def test_positive_delete_activation_key_2(self, description):
```
Comment (oshtaier) [inline tests/foreman/ui/test_activationkey.py]: That scenario seems as pretty redundant and can be easily verified in scope of `test_positive_delete_activation_key_with_different_name`

**Comment by elyezer** [comment]
ACK

**Comment by sthirugn** [comment]
ACK

**Pull request merged**